activesupport/test/message_verifier_test.rb

```diff
@@ -36,13 +38,25 @@ def test_simple_round_tripping
   end
 
   def test_verified_returns_false_on_invalid_message
-    assert !@verifier.verified("purejunk")
+    message = @verifier.generate(@data)
+    mock = MiniTest::Mock.new
+    mock.expect(:call, false, [message])
+    @verifier.stub(:valid_message?, mock) do
+      assert !@verifier.verified(message)
+    end
+    mock.verify
```

Comment:
I added the `expect` so `verified` could be tested based on the `valid_message?` behavior. It seems to me that without the mock, we would have to duplicate cases that `valid_message?` is being tested against in `test_valid_message`.